crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -474,44 +498,54 @@ impl<TX: DbTxMut + DbTx> DatabaseProvider<TX> {
                 }
             };
 
-            reverts
-                .entry(block_number)
+            revert_storage
+                .entry(block_and_address)
                 .or_default()
-                .entry(address)
-                .or_default()
-                .1
-                .push(old_storage);
+                .push((old_storage.key.into(), old_storage.value));
+
+            bundle_builder = bundle_builder
+                .state_storage(address, state_storage.entry(address).or_default().clone());
+            bundle_builder = bundle_builder.revert_storage(
+                block_number,
+                address,
+                revert_storage.entry(block_and_address).or_default().clone(),
```

Comment:
Same idea as [above ](https://github.com/paradigmxyz/reth/pull/8264/files#r1600570168)to avoid clone.